From 7e1f42417dab20d470d1e45dfa73d00c763d792d Mon Sep 17 00:00:00 2001
From: Paulo Flabiano Smorigo <pfsmorigo@br.ibm.com>
Date: Wed, 19 Sep 2012 20:50:38 -0300
Subject: [PATCH] increase the ieee1275 device path buffer size

There are cases when the openfirmware device path is bigger then 64 chars.

This should fix this bugzilla:
https://bugzilla.redhat.com/show_bug.cgi?id=857936
---
 grub-core/kern/ieee1275/init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/kern/ieee1275/init.c b/grub-core/kern/ieee1275/init.c
index 7d03a8a..5c45947 100644
--- a/grub-core/kern/ieee1275/init.c
+++ b/grub-core/kern/ieee1275/init.c
@@ -82,7 +82,7 @@ void (*grub_ieee1275_net_config) (const char *dev,
 void
 grub_machine_get_bootlocation (char **device, char **path)
 {
-  char bootpath[64]; /* XXX check length */
+  char bootpath[256]; /* Max device path length */
   char *filename;
   char *type;
    
-- 
1.7.10.4

